🐛 Fix issue - kg-bookmark tags available on RSS feed XML

pr3d opened the pull request:
fixes #11705

This will replace the bookmark card with the html anchor element.
To do so, it

- greps the `href` attribute to get the url
- greps the text content of the element with the class `kg-bookmark-publisher` to get the site title
- finally replaces the element with the class `kg-bookmark-card` with the html anchor element

pr3d committed:
:bug: Fix issue - kg-bookmark tags available on RSS feed XML

fixes #11705

This will replace the bookmark card with the html anchor element.
To do so, it

- greps the `href` attribute to get the url
- greps the text content of the element with the class `kg-bookmark-publisher` to get the site title
- finally replaces the element with the class `kg-bookmark-card` with the html anchor element

pr3d committed:
Merge branch 'master' of https://github.com/TryGhost/Ghost

kevinansfield reviewed (changes requested):
Hey @pr3d 👋 Thanks for looking at this! I've reviewed the code quickly and I think there are two things that need addressing:

1. The code currently assumes there is only one bookmark card in a post but there can in fact be many - we'll want to use cheerio to query for all bookmark cards and loop over and replace each one
2. The `<a>` that replaces each bookmark card should be wrapped in a `<p>` so that semantics and layout are maintained

pr3d committed:
Merge branch 'master' of https://github.com/TryGhost/Ghost

pr3d committed:
Fix issue - kg-bookmark tags available on RSS feed XML

This will update the previous code and implement the changes from kevinansfield's comment.
Pull Request: #11822

matthanley commented:
ping @kevinansfield for review 😄

daniellockyer commented:
Hey @pr3d! 👋🏻 Unfortunately your PR got automatically closed when we renamed branches yesterday because your fork seems to have been deleted. If you fancy reopening the PR again with a new fork, we'd be more than happy to take another look 🙂
